rfc/src/rfcs/0008-line-coverage.md

```diff
@@ -0,0 +1,156 @@
+- **Feature Name:** Line coverage (`line-coverage`)
+- **Feature Request Issue:** <https://github.com/model-checking/kani/issues/2610>
+- **RFC PR:** <https://github.com/model-checking/kani/pull/2609>
+- **Status:** Under Review
+- **Version:** 0
+- **Proof-of-concept:** <https://github.com/model-checking/kani/pull/2609>
+
+-------------------
+
+## Summary
+
+Add verification-based line coverage reports to Kani.
+
+## User Impact
+
+Nowadays, users can't easily obtain verification-based coverage reports in Kani.
+Generally speaking, coverage reports show which parts of the code under verification are covered and which are not.
+Because of that, coverage is often seen as a great metric to determine the quality of a verification effort.
+
+Moreover, some users prefer using coverage information for harness development and debugging.
+That's because coverage information provides users with more familiar way to interpret verification results.
+
+This RFC proposes adding a new option for verification-based line coverage reports to Kani.
+As mentioned earlier, we expect users to employ this coverage-related option on several stages of a verification effort:
+ * **Learning:** New users are more familiar with coverage reports than property-based results.
+ * **Development:** Some users prefer coverage results to property-based results since they are easier to interpret.
+ * **Debugging:** Users may find coverage reports particularly helpful when inputs are over-constrained (missing some corner cases).
+ * **Evaluation:** Users can easily evaluate where and when more verification work is needed (some projects aim for 100% coverage).
+
+Moreover, adding this option directly to Kani, instead of relying on another tools, is likely to:
+ 1. Increase the speed of development
+ 2. Improve testing for coverage features
+Which translates into faster and more reliable coverage options for users.
+
+## User Experience
```

Comment:
Can you add an initial paragraph explaining that this is still a work in progress? What about something like:

```suggestion
## User Experience

The goal is for Kani to generate code coverage report per harness in a well established format, such as LCOV, and possibly a summary in the output.
For now, we will focus on an interim solution that will enable us to assess our instrumentation and enable VSCode plugin integration.
```

Reply:
Thanks for the suggestion, I've added this with some minor changes.